src/broker/__tests__/reauthenticate.spec.js

```diff
@@ -0,0 +1,118 @@
+const {
+  saslOAuthBearerConnectionOpts,
+  newLogger,
+  describeIfOauthbearerEnabled,
+  createConnection,
+} = require('testHelpers')
+
+const { SaslAuthenticate: apiSaslAuthenticateKey } = require('../../protocol/requests/apiKeys')
+const Broker = require('../index')
+
+/**
+ * Session lifetime on broker is configured in docker-compose.2_4_oauthbearer.yml with 15000ms in env KAFKA_CONNECTIONS_MAX_REAUTH_MS
```

Comment:
Is it possible to reauthenticate much sooner than this? I'm thinking maybe we can avoid the excessive sleeps in the test by either setting the `reauthenticationThreshold` to like 14900ms or something so we only need to sleep a little bit, or override `broker.sessionLifetime` to be 0 whenever we want to trigger a reauthentication. Something like:

```ts
    const connection = createConnection(saslOAuthBearerConnectionOpts())
    const spy = jest.spyOn(connection, 'send')
    broker = new Broker({
      connection,
      logger: newLogger(),
    })
    await broker.connect()
    await broker.listGroups()
    await broker.listGroups()

    broker.sessionLifetime = 0n;
    await broker.listGroups()

    expect(getAuthCalls(spy).length).toBe(2)
```

With all the sleeps in the current test, we're adding about half a minute or more to the execution time that we can maybe avoid.

Reply:
yes sure, I will reduce it by using higher `reauthenticationThreshold `.